src/go/k8s/pkg/resources/statefulset.go

```diff
@@ -76,8 +81,10 @@ func (r *StatefulSetResource) Ensure(ctx context.Context) error {
 	r.LastObservedState = &sts
 
 	// Ensure StatefulSet #replicas equals cluster requirement.
-	if sts.Spec.Replicas != r.pandaCluster.Spec.Replicas {
+	if sts.Spec.Replicas != nil && r.pandaCluster.Spec.Replicas != nil && *sts.Spec.Replicas != *r.pandaCluster.Spec.Replicas {
+		logger.Info(fmt.Sprintf("StatefulSet %s has replicas set to %d but need %d. Going to update", r.Key().Name, *sts.Spec.Replicas, *r.pandaCluster.Spec.Replicas))
```

Comment:
This detailed information shouldn't be logged on Info. Could we use the `log.V(3).Info` here? We should follow [kubernetes logging conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/logging.md).

Reply:
this is critical information to debug what the controller is doing, this will be logged only when the scaling really happens. Why do you think it won't fit into info?

Reply:
Sorry I understood it wrongly. You are right that it will happen only on updated. This information wouldn't cuddle the logs.